tests/statefulprecompiles/allow_list_test.go

```diff
@@ -0,0 +1,284 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package statefulprecompiles
+
+import (
+	"testing"
+
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/subnet-evm/core/rawdb"
+	"github.com/ava-labs/subnet-evm/core/state"
+	"github.com/ava-labs/subnet-evm/precompile"
+	"github.com/ava-labs/subnet-evm/precompile/allowlist"
+	"github.com/ava-labs/subnet-evm/vmerrs"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAllowListRun(t *testing.T) {
+	type test struct {
+		caller      common.Address
+		input       func() []byte
+		suppliedGas uint64
+		readOnly    bool
+
+		expectedRes []byte
+		expectedErr string
+
+		config *allowlist.AllowListConfig
+
+		assertState func(t *testing.T, state *state.StateDB)
+	}
```

Comment:
AllowListConfig does not implement StatefulPrecompileConfig.
Currently allow_list_test tests it's own functionality and other precompiles do not need to test them in their files. 
We can complicate these UTs a little bit to use same test suite and run same tests for every precompile. It would be some repetitive work and would test the same code multiple times. Currently I removed those repeated tests from other files and only kept them in contract/tx allow lists since they do not have any other tests.

Reply:
I think it would be worth it to duplicate the tests to ensure that the behavior stays the same for the allow list across each one.

Since this is very important to ensure the behavior stays consistent, I think it's worthwhile. Should this be a follow up PR or GitHub issue to follow up with?

Reply:
I prefer a follow up not to block this further. Created:https://github.com/ava-labs/subnet-evm/issues/497